.github/workflows/release.yml

```diff
@@ -0,0 +1,111 @@
+        draft: true
+        prerelease: false
+  build:
+    name: Build application
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        target: [macos, windows, linux, apk]
+        include:
+        - os: macos-10.15
+          target: macos
+          path: build/macos/Build/Products/Release
+        - os: windows-2019
+          target: windows
+          path: build\windows\runner\Release
+        - os: ubuntu-18.04
+          target: linux
+          path: build/linux/release/bundle
+        - os: ubuntu-18.04
+          target: apk
+          path: build/app/outputs/apk/release/
+      # Disable fail-fast as we want results from all even if one fails.
+      fail-fast: false
+    needs: create-release
+    steps:
+    # Set up Flutter and add it to the path.
+    - name: Clone Flutter repository with master channel
+      uses: actions/checkout@v2
+      with:
+        repository: 'flutter/flutter'
+        ref: 'master'
+        path: 'flutter'
+        fetch-depth: 0
+    - name: Add Flutter to the PATH for Linux and macOS
+      if: startsWith(matrix.os, 'macOS') || startsWith(matrix.os, 'ubuntu')
+      run: echo "::add-path::$GITHUB_WORKSPACE/flutter/bin"
+    - name: Add Flutter to the PATH for Windows
+      if: startsWith(matrix.os, 'windows')
+      run: echo "::add-path::${env:GITHUB_WORKSPACE}\flutter\bin"
+    - name: Install Linux dependencies
+      if: startsWith(matrix.target, 'linux')
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y libgtk-3-dev libx11-dev pkg-config cmake ninja-build libblkid-dev
+    - name: Install Android dependencies
+      if: startsWith(matrix.target, 'apk')
+      uses: actions/setup-java@v1
+      with:
+        java-version: '12.x'
+    - name: Enable desktop support
+      run: |
+        flutter config --enable-linux-desktop
+        flutter config --enable-macos-desktop
+        flutter config --enable-windows-desktop
+    - name: Run Flutter doctor
+      run: flutter doctor -v
+
+    # Clone the Gallery repository under `code`, to avoid conflicts with `flutter`.
+    - name: Clone the Gallery repository
+      uses: actions/checkout@v2
+      with:
+        path: 'code'
+    - name: Get packages for the Flutter project
+      run: flutter pub get
+      working-directory: code
+
+    # Build the application for desktop.
+    - name: Build project
+      run: |
+        flutter create .
+        flutter build -v ${{ matrix.target }} --release
+      working-directory: code
+    - name: Compress build for Linux and macOS
+      if: startsWith(matrix.os, 'macOS') || startsWith(matrix.os, 'ubuntu')
+      run: zip -r $GITHUB_WORKSPACE/flutter_gallery_${{ matrix.target }} .
+      working-directory: code/${{ matrix.path }}
+    - name: Compress build for Windows
+      if: startsWith(matrix.os, 'windows')
+      run: compress-archive -Path * -DestinationPath ${env:GITHUB_WORKSPACE}\flutter_gallery_${{ matrix.target }}.zip
```

Comment:
Not that this won't run for all users; see https://github.com/flutter/flutter/wiki/Desktop-shells#distributing

If a zip is the distribution method the C++ redistributables should be included in it (i.e., copied into this folder before making the archive).

Reply:
Makes sense!

@stuartmorgan I'm adding them here: https://github.com/flutter/gallery/pull/320